### DISCUSSION TITLE
Nest Wildcard Exclusion?

### DISCUSSION THREAD

**Discussion by matthewjchou** [fluent/fluent-bit#4934] [Q&A]
Can you use Wildcard in Nest to exclude a fields?
I have a bunch of keys that fall into 2 categories, set A and set B. I'd like to nest set A without nesting set B, but set A doesn't have common prefixes and set B does.  
Ex.
```
{
 a: aa,
 b: bb,
 common_c: cc,
 common_d: dd
}
```
I've tried different regex solutions, but I don't think WILDCARD uses fully featured regex - although I can't find what it actually does use.
Ex.
```
[FILTER]
    name nest
    wildcard ^((?!common).)*$
    operation nest
    nest_under sd
```
With the expected output of:
```
{
 sd: {
    a: aa,
    b: bb
 },
 common_c: cc,
 common_d: dd
}
```
Any help would be appreciated - also open to alternative solutions.

**Comment by patrick-stephens** [comment]
I'd probably just drop the fields you don't want either before or after in a separate modify filter. It keeps things simple, particularly if you modify the fieldset in the future in a way that impacts your regex.

My general policy is to try to avoid complex regexes: they're expensive if only in having to maintain them for my cognitive load!
